server/src/main/java/org/opensearch/index/mapper/MatchOnlyTextFieldMapper.java

```diff
@@ -0,0 +1,310 @@
+            NamedAnalyzer searchAnalyzer = analyzers.getSearchAnalyzer();
+            NamedAnalyzer searchQuoteAnalyzer = analyzers.getSearchQuoteAnalyzer();
+
+            if (fieldType.indexOptions().compareTo(IndexOptions.DOCS) > 0) {
+                throw new IllegalArgumentException("Cannot set position_increment_gap on field [" + name + "] without positions enabled");
+            }
+            if (positionIncrementGap.get() != POSITION_INCREMENT_GAP_USE_ANALYZER) {
+                if (fieldType.indexOptions().compareTo(IndexOptions.DOCS) < 0) {
+                    throw new IllegalArgumentException(
+                        "Cannot set position_increment_gap on field [" + name + "] without indexing enabled"
+                    );
+                }
+                // for index analyzer we don't set positionIncrementGap whereas for search analyzer its set because
+                // phrase queries, which make use of it, should work fine as they will directly work on the field value
+                // per matched document by reading from _source field.
+                indexAnalyzer = new NamedAnalyzer(indexAnalyzer, positionIncrementGap.get());
+                searchAnalyzer = new NamedAnalyzer(searchAnalyzer, positionIncrementGap.get());
+                searchQuoteAnalyzer = new NamedAnalyzer(searchQuoteAnalyzer, positionIncrementGap.get());
+            }
+            TextSearchInfo tsi = new TextSearchInfo(fieldType, similarity.getValue(), searchAnalyzer, searchQuoteAnalyzer);
+            MatchOnlyTextFieldType ft = new MatchOnlyTextFieldType(
+                buildFullName(context),
+                index.getValue(),
+                fieldType.stored(),
+                tsi,
+                meta.getValue()
+            );
+            ft.setIndexAnalyzer(indexAnalyzer);
+            ft.setEagerGlobalOrdinals(eagerGlobalOrdinals.getValue());
+            ft.setBoost(boost.getValue());
+            if (fieldData.getValue()) {
+                ft.setFielddata(true, freqFilter.getValue());
+            }
+            return ft;
+        }
+
+        @Override
+        protected List<Parameter<?>> getParameters() {
+            return Arrays.asList(
+                index,
+                store,
+                indexOptions,
+                norms,
+                termVectors,
+                analyzers.indexAnalyzer,
+                analyzers.searchAnalyzer,
+                analyzers.searchQuoteAnalyzer,
+                similarity,
+                positionIncrementGap,
+                fieldData,
+                freqFilter,
+                eagerGlobalOrdinals,
+                indexPhrases,
+                indexPrefixes,
+                boost,
+                meta
+            );
+        }
+    }
+
+    /**
+     * The specific field type for MatchOnlyTextFieldMapper
+     *
+     * @opensearch.internal
+     */
+    public static final class MatchOnlyTextFieldType extends TextFieldMapper.TextFieldType {
```

Comment:
```suggestion
    public static final class MatchOnlyTextFieldType extends TextFieldType {
```
To be consistent with other types.

Reply:
Done